File: apps/docs/src/code/python/basics/process_stream_stdout.py
from e2b import Session

session = Session.create(
    id="Nodejs",
    on_stdout=lambda output: print("session", output.line),  # $HighlightLine
)

proc = session.process.start('echo "Hello World!"')
proc.wait()
# output: session Hello World!

proc_with_custom_handler = session.process.start(
    'echo "Hello World!"',
    on_stdout=lambda output: print("process", output.line),  # $HighlightLine
)
proc_with_custom_handler.wait()
# output: process Hello World!

session.close()

File: apps/docs/src/code/js/basics/process_stream_stdout.js
import { Sandbox } from '@e2b/sdk'

const sandbox = await Sandbox.create({
  id: 'base',
  onStdout: (output) => console.log('sandbox', output.line), // $HighlightLine
})

const proc = await sandbox.process.start({
  cmd: 'echo "Hello World!"',
})
await proc.wait()
// output: sandbox Hello World!

const procWithCustomHandler = await sandbox.process.start({
  cmd: 'echo "Hello World!"',
  onStdout: (data) => console.log('process', data.line), // $HighlightLine
})
await procWithCustomHandler.wait()
// output: process Hello World!

await sandbox.close()
